pyccl/halos/hmfunc.py

```diff
@@ -464,21 +485,22 @@ def _setup(self, cosmo):
                         -0.301, -0.301, -0.319, -0.336])
 
         ldelta = np.log10(delta)
-        self.pA0 = interp1d(ldelta, alpha)
-        self.pa0 = interp1d(ldelta, eta)
-        self.pb0 = interp1d(ldelta, beta)
-        self.pc0 = interp1d(ldelta, gamma)
-        self.pd0 = interp1d(ldelta, phi)
+        extrap_kw = {"bounds_error": False, "fill_value": "extrapolate"}
```

Comment:
We want this to fail if a user passes values for `ldelta` beyond the Tinker interpolation ranges, since things are not tested in that regime.

Reply:
The tests are failing for some reason though. There is one test that uses a value slightly outside of the range. And these are tests that were already there before I make any changes. I'll get back to you on that.

Reply:
I've reverted those changes.

Reply:
Update on this: I discovered the reason I was getting interpolation-out-of-range errors. In `Cosmology`, the default halo model mass definition is virial. For some concentrations, the Delta-translation of the virial mass definition is below 200, which is the lower interpolation boundary in Tinker08 and Tinker10.
This normally wouldn't pass the tests, but it was left uncovered. In this PR I increased coverage, hence the tests were failing.

What I'm going to do is change the halo model default mass definition from virial to 200m, which should solve the issue.

PS: I know the the halo model non-linear power spectrum is deprecated, but I tried to save it by moving the relevant parameters to `extra_parameters`.